bots/mainbot/pilgrim.js

```diff
@@ -122,66 +127,96 @@ pilgrim.takeTurn = (self) => {
-				return pilgrim.move(self, pilgrim.home);
+				let t = {x: pilgrim.home.x + 1, y: pilgrim.home.y}
+				utilities.log(self, `Moving from (${self.me.x}, ${self.me.y}) toward home at (${t.x}, ${t.y}) - adjacent? ${utilities.isAdjacent(self.me, pilgrim.home)}`);
+				// utilities.log(self, `Moving from (${self.me.x}, ${self.me.y}) toward home at (${pilgrim.home.x}, ${pilgrim.home.y}) - adjacent? ${utilities.isAdjacent(self.me, pilgrim.home)}`);
+				// return pilgrim.move(self, pilgrim.home);
+				return pilgrim.move(self, t);
 			}
 	}
 }
 
 // Move toward a target.
 pilgrim.move = (self, target) => {
-	let path = undefined;
-
-	try {
-		utilities.log(self, `Target: (${target.x}, ${target.y})`);
-		path = movement.moveTo(self, target.x, target.y);
-		utilities.log(self, `Path length: ${path.length}`);
-	} catch(e) {
-		utilities.log(self, "Path-finding algorithm raised an exception! Reverting to random movement.");
-		return pilgrim.random_move(self);
+	if (!pilgrim.path) {
+		try {
+			utilities.log(self, `Calculating path to: (${target.x}, ${target.y})`);
+			pilgrim.path = movement.moveTo(self, target.x, target.y);
+			utilities.log(self, `Path length: ${	pilgrim.path.length}`);
+		} catch(e) {
+			utilities.log(self, "Movement exception: " + e);
+			utilities.log(self, "Path-finding algorithm raised an exception! Reverting to random movement.");
+			return pilgrim.random_move(self);
+		}
 	}
-	let i = 0;
 
-	while (i < path.length && !utilities.isOpen(self, path[i])) {
-		utilities.log(self, `Step ${i}: (${path[i].x}, ${path[i].y})`)
-		i++;
-	}
+	// Path returned is empty
+	if (pilgrim.path.length === 0) {
+		if (utilities.inMovementRange(self, target)) {
+			// Target in movement range, so we can move one square towards target
+			let dx = target.x - self.x;
+			let dy = target.y - self.y;
+			let step = {x: self.x, y: self.y};
+
+			if (dx > 0) {
+				step.x = 1
+			} else if (dx < 0) {
+				step.dx = -1;
+			}
 
-	if (i >= path.length) {
-		utilities.log(self, "Failed to find a path! Reverting to random movement.");
-		return pilgrim.random_move(self);
+			if (dy > 0) {
+				step.y = 1;
+			} else if (dy < 0) {
+				step.y = -1;
+			}
+
+			pilgrim.path = [step];
+			utilities.log("Closing in on target. Moving one tile toward target.");
+
+		} else {
+			// No valid path to target
+			utilities.log(self, "Failed to find a path! Reverting to random movement.");
+			pilgrim.path = undefined;
+			return pilgrim.random_move(self);
+		}
 	}
 
-	utilities.log(self, "Distance to step: " + utilities.getDistance(self.me, path[i]));
-	utilities.log(self, "Max speed: " + SPECS.UNITS[SPECS.PILGRIM].SPEED);
-	if (utilities.getDistance(self.me, path[i]) > SPECS.UNITS[SPECS.PILGRIM].SPEED) {
+	let step = pilgrim.path.shift();
+
+	utilities.log(self, `Stepping to: (${step.x}, ${step.y})`)
+	utilities.log(self, "Distance to step: " + utilities.getDistance(self.me, step));
+	if (utilities.getDistance(self.me, step) > SPECS.UNITS[SPECS.PILGRIM].SPEED) {
```

Comment:
So I think this happens when the bot tries to take a step that a robot moved into the path. The movement fails because the square is occupied but the path doesn't get reset, so when you try to move to the next square in the path it is too far. Not sure how to handle this other than recalculate the path.